definitions/features/katello_service.rb

```diff
@@ -27,13 +29,65 @@ def make_start(spinner, options = {})
       spinner.update 'No katello service to start'
     else
       filters = "--only #{services.join(',')}"
-      spinner.update 'Starting the katello services..'
+      spinner.update 'Starting katello services..'
       execute!("katello-service start #{filters}")
     end
   end
 
+  def restart(spinner, options = {})
+    spinner.update 'Restarting katello services..'
+    if options[:only] || options[:exclude]
+      filters = construct_filters_for_restart(options)
+      execute!("katello-service restart #{filters}")
+    else
+      execute!('katello-service restart')
+    end
+  end
+
+  def run_with_hammer_ping_retry(spinner, action = '', services_options = {})
+    meth_call_by_action(action, spinner, services_options)
+    spinner.update 'Sometimes, services will take time to be ready'
+    begin
+      retry_hammer_ping(spinner)
+    rescue StandardError => e
+      logger.error e.message
+    end
+  end
+
   private
 
+  def retry_hammer_ping(spinner)
+    RETRIES_FOR_SERVICES_RESTART.times do |retry_count|
+      spinner.update "Retry #{retry_count + 1}: checking status by hammer ping"
```

Comment:
Please include the number max attempts as well and use `Try` instead of `Retry` as the first attempt is not effectively a retry :)

```ruby
      spinner.update "Try #{retry_count + 1}/#{RETRIES_FOR_SERVICES_RESTART}: checking status by hammer ping"
```